README.md
# Term Style :dancer:

[![Build Status](https://travis-ci.com/iamtheblackunicorn/TermStyle.svg?branch=main)](https://travis-ci.com/iamtheblackunicorn/TermStyle)

*Making your Dart command-line apps funky! :dancer:*

## About :books:

I always thought that the output of command-line apps was just a slew of text and a little bit boring. So I decided to spice things up a little! Term Style gives you the ability to use emojis and colors in your terminal!

## What's available?

### Colors

In terms of colors, these colors are currently provided by Term Style:

- Red
- Green
- Blue
- Cyan
- Magenta
- Yellow

### Emojis

In terms of emojis, these emojis are currently provided by Term Style in four different categories:

#### Faces
- `grinningFace`: :grinning:
- `laughCry`: :joy:
- `upsideDownFace`: :upside_down_face:
- `heartsInFace`: :smiling_face_with_three_hearts:
- `smilingFace`: :blush:
- `haloFace`: :innocent:
- `winkingFace`: :wink:
- `kissingFace`: :kissing_closed_eyes:
- `angryFace`: :angry:
- `crazyFace`: :zany_face:
- `thinkingFace`: :thinking:
- `naughtyface`: :smiling_imp:

#### Animals
- `seeNoEvil`: :see_no_evil:
- `animalGorilla`: :gorilla:
- `windCloud`: :dash:
- `sweatDroplets`: :sweat_drops:
- `dogFace`: :dog:
- `unicornHead`: :unicorn:
- `catFace`: :cat:
- `horseHead`: :horse:
- `pandaFace`: :panda_face:
- `bearFace`: :bear:
- `tigerFace`: :tiger:

#### Symbols
- `readHeart`: :heart:
- `blackHeart`: :black_heart:
- `backArrow`: :back:
- `soonArrow`: :soon:
- `topArrow`: :top:
- `infinitySign`: :infinity:
- `multiplySign`: :heavy_multiplication_x:
- `plusSign`: :heavy_plus_sign:
- `divideSign`: :heavy_division_sign:
- `minusSign`: :heavy_minus_sign:
- `tickSign`: :heavy_check_mark:

#### Tools

- `kitchenKnife`: :knife:
- `scissors`: :scissors:
- `bomb`: :bomb:
- `gun`: :gun:
- `hammer`: :hammer:
- `wrench`: :wrench:
- `hammerAndWrench`: :hammer_and_wrench:
- `hammerAndPick`: :hammer_and_pick:
- `pick`: :pick:
- `woodSaw`: `not supported by GitHub markdown`
- `axe`: :axe:

## Installation :inbox_tray:

### Adding to your project

To add TermStyle to your project's dependencies, add this to your project's `pubspec.yaml`:

#### From *GitHub*

```YAML
dependencies:
  ...
  termstyle:
    git: git://github.com/iamtheblackunicorn/TermStyle.git
```

#### From *Pub.dev*

```YAML
dependencies:
  ...
  termstyle: ^1.1.0
```

The three dots represent anything else that you might have in the `dependencies` section.
Having done that, re-fetch your project's dependencies by running this in the project's root directory:

```bash
$ dart pub get
```

## Usage :hammer:

### Importing

To import the color module into your code, use this line:

```dart
import 'package:termstyle/colors.dart';
```

To import the emoji module into your code, use this line:

```dart
import 'package:termstyle/emojis.dart';
```

### API

#### Emojis

##### Map<String, dynamic> emojiDataBase()

This method returns a database of 44 select emojis in form of a map in four select categories.

##### String getEmoji(String emoji)

This method returns an emoji by its name as a string.

##### void printEmoji(String emoji)

This method prints an emoji by its name to the console.

##### void testAllEmojis()

This method iterates over the emoji database, once getting and printing a string by the name of the emoji, and once by directly printing a string by the name of the emoji to the console.

#### Colors

##### Map<String, dynamic> termColors()

This method is a database of all the six colors for pretty console colors. It returns a map.

##### String getColoredString(String message, String userColor)

This method returns a string in the specified color.

##### void printColoredString(String message, String userColor)

This method prints a string in the specified color to the console.

##### void testAllColors()

This method tests all available colors.

## Note :scroll:

- TermStyle by Alexander Abraham a.k.a. *The Black Unicorn*
- licensed under the MIT license


CHANGELOG.md
# CHANGELOG

## 1.0.0: Initial release

- initial release
- upload to pub.dev
- upload to GitHub
- 6 colors
- 33 emojis from three different categories

## 1.1.0: The Tools update

- added a fourth section, tools!
- fixed a typo with the `redHeart` emoji
- added docs


lib/colors.dart
/*
TermStyle by Alexander Abraham a.k.a. The Black Unicorn.
Licensed under the MIT license.
TermStyle's Color API contains a selection of 6 colors.
*/

/// This method is a database of all
/// the six colors for pretty console
/// colors. It returns a map.
Map<String, dynamic> termColors() {
  Map<String, dynamic> colors = {
    'red': '\x1B[31m',
    'green': '\x1B[32m',
    'blue': '\x1B[34m',
    'cyan': '\x1B[36m',
    'magenta': '\x1B[35m',
    'yellow': '\x1B[33m',
    'suffix': '\x1B[0m',
  };
  return colors;
}

/// This method returns a string in the specified color.
String getColoredString(String message, String userColor) {
  Map<String, dynamic> colors = termColors();
  String color = colors[userColor];
  String suffix = colors['suffix'];
  String userMessage = '$color$message$suffix';
  return userMessage;
}

/// This method prints a string in the specified color
/// to the console.
void printColoredString(String message, String userColor) {
  Map<String, dynamic> colors = termColors();
  String color = colors[userColor];
  String suffix = colors['suffix'];
  String userMessage = '$color$message$suffix';
  print(userMessage);
}

/// This method tests all available colors.
void testAllColors() {
  List<String> allColors = [
    'red',
    'green',
    'blue',
    'cyan',
    'magenta',
    'yellow'
  ];
  for (int i = 0; i < allColors.length; i++) {
    String message = 'Hello World!';
    String fetchedColor = getColoredString(message, allColors[i]);
    print(fetchedColor);
    printColoredString(message, allColors[i]);
  }
}


lib/emojis.dart
/*
TermStyle by Alexander Abraham a.k.a. The Black Unicorn.
Licensed under the MIT license.
TermStyle's Emoji API contains a selection of 33 emojis.
*/

/// This method returns a database of 44 select emojis
/// in form of a map in four select categories.
Map<String, dynamic> emojiDataBase() {
  Map<String, dynamic> emojidb = {
    // faces
    'grinningFace': '\u{1f600}',
    'laughCry': '\u{1f602}',
    'upsideDownFace': '\u{1f643}',
    'heartsInFace': '\u{1f970}',
    'smilingFace': '\u{1f60a}',
    'haloFace': '\u{1f607}',
    'winkingFace': '\u{1f609}',
    'kissingFace': '\u{1f61a}',
    'angryFace': '\u{1f620}',
    'crazyFace': '\u{1f92a}',
    'thinkingFace': '\u{1f914}',
    'naughtyface': '\u{1f608}',

    // animals
    'seeNoEvil': '\u{1f648}',
    'animalGorilla': '\u{1f98d}',
    'windCloud': '\u{1f4A8}',
    'sweatDroplets': '\u{1f4A6}',
    'dogFace': '\u{1f436}',
    'unicornHead': '\u{1f984}',
    'catFace': '\u{1f431}',
    'horseHead': '\u{1f434}',
    'pandaFace': '\u{1f43c}',
    'bearFace': '\u{1f43b}',
    'tigerFace': '\u{1f42f}',

    // symbols
    'redHeart': '\u{2764}',
    'blackHeart': '\u{1f5a4}',
    'backArrow': '\u{1f519}',
    'soonArrow': '\u{1f51c}',
    'topArrow': '\u{1f51d}',
    'infinitySign': '\u{267e}',
    'multiplySign': '\u{2716}',
    'plusSign': '\u{2795}',
    'divideSign': '\u{2797}',
    'minusSign': '\u{2796}',
    'tickSign': '\u{2714}',

    // tools
    'kitchenKnife': '\u{1f52a}',
    'scissors': '\u{2702}',
    'bomb': '\u{1f4a3}',
    'gun': '\u{1f52b}',
    'hammer': '\u{1f528}',
    'wrench': '\u{1f527}',
    'hammerAndWrench': '\u{1f6e0}',
    'hammerAndPick': '\u{2692}',
    'pick': '\u{26cf}',
    'woodSaw': '\u{1fa9a}',
    'axe': '\u{1fa93}'
  };
  return emojidb;
}

/// This method returns an emoji by its name as
/// a string.
String getEmoji(String emoji) {
  Map<String, dynamic> emojiDB = emojiDataBase();
  String userEmoji = emojiDB[emoji];
  return userEmoji;
}

/// This method prints an emoji by its name
/// to the console.
void printEmoji(String emoji) {
  Map<String, dynamic> emojiDB = emojiDataBase();
  String userEmoji = emojiDB[emoji];
  print(userEmoji);
}

/// This method iterates over the emoji database,
/// once getting and printing a string by the name of the emoji,
/// and once by directly printing a string by the name of the emoji
/// to the console.
void testAllEmojis() {
  Map<String, dynamic> emojiDBOne = emojiDataBase();
  emojiDBOne.keys.forEach((k) => print(getEmoji('$k')));
  Map<String, dynamic> emojiDBTwo = emojiDataBase();
  emojiDBTwo.keys.forEach((k) => printEmoji('$k'));
}
